src/aosm/azext_aosm/build_processors/base_processor.py

```diff
@@ -201,11 +201,13 @@ def _generate_schema(
                     )
                 else:
                     if prop in default_values:
-                        # AOSM wants null as a string
-                        # TODO: Flag with RP that this is a bug?
+                        # RP does not accept null values as defaults,
+                        # which will fail the NFDV publish
+                        # So if there is null default, add to required

```

Comment:
`# So if there is null default, add to required`
The code tells us this, what would be more useful is _why_ we're doing it
```suggestion
                        # To workaround this we don't set a default, so must add to required params instead

```